src/main/java/org/openmrs/addonindex/service/ElasticSearchIndex.java

```diff
@@ -100,16 +100,21 @@ public void index(AddOnInfoAndVersions infoAndVersions) throws IOException {
 			boolQB.filter(QueryBuilders.matchQuery("type", type));
 		}
 		if (query != null) {
-			boolQB.should(QueryBuilders.matchQuery("name", query).boost(4.0f));
-			boolQB.should(QueryBuilders.prefixQuery("_all", query).boost(1.5f));
-			boolQB.should(QueryBuilders.matchPhrasePrefixQuery("_all", query).slop(2).fuzziness("AUTO"));
+			//Following clause return a module whose UID matches exactly
+			boolQB.should(QueryBuilders.termQuery("_id", query).boost(1500.0f));
+			boolQB.should(QueryBuilders.prefixQuery("name", query).boost(4.0f));
+			boolQB.should(QueryBuilders.termQuery("tags", query).boost(1500.0f));
```

Comment:
When an exact uid matches then it never shows any other result. Similarly when a tag matches it's never possible for a UID to match and hence we do not have to worry about them clashing. Either way since you have mentioned it, I shall change it ☺️